kernel/mmanutils.c

```diff
@@ -536,6 +579,52 @@ int myst_release_process_mappings(pid_t pid)
     return ret;
 }
 
+int proc_pid_maps_vcallback(myst_buf_t* vbuf)
+{
+    int ret = 0;
+    pid_t pid = myst_getpid();
+    struct locals
+    {
+        char maps_entry[PATH_MAX];
+    }* locals = NULL;
+
+    if (!(locals = malloc(sizeof(struct locals))))
+        ERAISE(-ENOMEM);
+
+    myst_buf_clear(vbuf);
+
+    myst_spin_lock(&_mappings_lock);
+    {
+        for (myst_process_mapping_t* p = _mappings; p; p = p->next)
+        {
+            if (p->pid == pid && p->pathname)
+            {
+                // ATTN: device and inode number are not reported
+                // shared or private perms bit is always marked 'p',
+                // as MAP_SHARED is not supported.
+                snprintf(
+                    locals->maps_entry,
+                    sizeof(locals->maps_entry),
+                    "%08lx-%08lx %c%c%cp %08lx 00:00 0 %s\n",
+                    (long)p->addr,
+                    (long)p->addr + p->size,
+                    p->prot & PROT_READ ? 'r' : '-',
+                    p->prot & PROT_WRITE ? 'w' : '-',
+                    p->prot & PROT_EXEC ? 'x' : '-',
+                    p->offset,
+                    p->pathname);
+                // Insert new entry at beginning
+                myst_buf_insert(
```

Comment:
Is it expected for the vcallback function to return 0 even if myst_buf_insert() failed because not enough space left in the buf?

Other procfs vcallback functions don't check myst_buf_xyz() function return either.

Reply:
It does make sense to propagate the error. Have added error checking to this callback and others.